### ISSUE TITLE
Make diagrams accessible

### ISSUE THREAD

**Issue by gkellogg** [opencreds/website#28]
Right now, there's a manual process of creating a textual description of a use case UML and using [PlantUML](http://www.plantuml.com/) to create an SVG. Future needs require making such diagrams accessible.

(from @halindrome)

**Comment by gkellogg** [comment]
One thought would be to automate the process by including the text in the document with a processing step, the extracts it, creates the SVG, and adds it back into the document, preserving the source as a longdesc.

**Comment by halindrome** [comment]
Yeah.  I am wary of relying upon another service for public facing documents.  I don't mind it when I am developing - if it went down I would know immediately.  So perhaps a hybrid solution where we do what you suggest, but create a static version for the public-facing version (automagically).

**Comment by gkellogg** [comment]
I wouldn't want to depend on a service, but perhaps a post-commit book which built the SVG on Travis and committed it back automatically.

In general, a generic way to run tasks on commit to perform many such tasks would be useful. We briefly explored this for doing link checking in CSVW, but manifest index building and TAR generation in rdf-tests would be useful too.

**Comment by halindrome** [comment]
Yes - I like the idea of a Travis hook.  There is a node.js solution that
can hit the plantuml.org service and get SVG back.   I will at it to my
todo list and see what I can mock up.

On Sun, Jan 31, 2016 at 1:01 PM, Gregg Kellogg notifications@github.com
wrote:

> I wouldn't want to depend on a service, but perhaps a post-commit book
> which built the SVG on Travis and committed it back automatically.
> 
> In general, a generic way to run tasks on commit to perform many such
> tasks would be useful. We briefly explored this for doing link checking in
> CSVW, but manifest index building and TAR generation in rdf-tests would be
> useful too.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/opencreds/website/issues/28#issuecomment-177584465.

## 

Shane McCarron
halindrome@gmail.com
